### PR TITLE
FIX: image respect norm limits w/ None

### LINKED ISSUE #10072: imshow doesn't properly display some images
### Bug report

**Bug summary**
When using imshow with larger images, you lose pixel resolution for low intensity pixels with lots of nearby zeros (not sure if the intensity or nearby zeros are relevant, or just relative isolation of the pixel). It's like they're being interpolated away, even if interpolation is set to nearest. This is new behavior in version 2.1.

**Code for reproduction**
Use the attached npy file, which is an image referenced in this test script.
```python
import matplotlib.pyplot as plt
import numpy as np

img = np.load('test_image.npy')

#Display full image
plt.imshow(img, interpolation = 'nearest')
plt.clim(0, 5)
plt.show()

#Display part of image
plt.imshow(img[200:400,500:700], interpolation = 'nearest')
plt.clim(0, 5)
plt.show()
```

**Actual outcome**
Using matplotlib 2.1.1:
Screenshot from the display of full image (first plot in code above):
full_plot_2p1

Using the  zoom tool to zoom in on a region when the full image is plotted:
zoom_full_plot_2p1

Plotting roughly the same zoom region initially (second plot in code above):
partial_plot_2p1

**Expected outcome**
It should all look like the third image. Here's the output when the same code is run in version 2.0.2
Screenshot from the display of full image (first plot in code above):
full_plot_2p0

Using the  zoom tool to zoom in on a region when the full image is plotted:
zoom_full_plot_2p0

Plotting roughly the same zoom region initially (second plot in code above):
partial_plot_2p0

**Matplotlib version**
  * Operating system: macOS 10.12.6
  * Matplotlib version: 2.1.1 (broken) and 2.0.2 (working)
  * Matplotlib backend (`print(matplotlib.get_backend())`): MacOSX
  * Python version: 2.7.12
  * Other libraries: numpy version 1.13.3

All libraries are installed through conda, using the standard channel. This includes both versions of matplotlib (2.1.1 and 2.0.2).

Numpy file with image data:
[test_image.zip](https://github.com/matplotlib/matplotlib/files/1579886/test_image.zip)

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Re-closes #10072 

We are slowly suqishing the "big number" bugs in imshow.  

This one was caused by only specifying `vmax` but not `vmin`.  Algorithm is the same as before, but if `vmin` is `None`, then we scale by `a_min`.  

```python
import matplotlib.pyplot as plt
import numpy as np

x=np.random.random([3,3])
x[1,1]=1.e80
plt.imshow(x,vmax=2)

plt.show()
```

#### Before
figure_1

#### After

figur

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -390,22 +390,24 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
...
-                    else:
-                        a_min = np.float64(newmin)
-                    newmax = vmid + dv * 1.e7
-                    if newmax > a_max:
-                        newmax = None
-                    else:
-                        a_max = np.float64(newmax)
-                    if newmax is not None or newmin is not None:
-                        A_scaled = np.clip(A_scaled, newmin, newmax)
+                vmin = self.norm.vmin if self.norm.vmin is not None else a_min
```
Comment (tacaswell) [inline lib/matplotlib/image.py]: I would do `if self.norm.vmin is None or self.norm.vmax is None: self.norm.autoscale_None(A)`

Comment (jklymak) [reply lib/matplotlib/image.py]: Sure - its OK to modify the norm here?

Comment (jklymak) [reply lib/matplotlib/image.py]: don't even need the if-statement in that case.

**Comment by jklymak** [comment]
Ping @tacaswell

**Comment by jklymak** [comment]
squashed and rebased....

**Comment by dstansby** [comment]
Does this want a backport to `2.2.x`?

**Comment by tacaswell** [comment]
backported to v2.2.x as faf9684bb

**Pull request merged**